mindsdb/integrations/handlers/eventstoredb_handler/eventstoredb_handler.py

```diff
@@ -0,0 +1,212 @@
+from mindsdb_sql.parser.ast.base import ASTNode
+from mindsdb_sql.parser.ast import Select
+from mindsdb.utilities import log
+from mindsdb.integrations.libs.base import DatabaseHandler
+from mindsdb.integrations.libs.response import (
+    HandlerStatusResponse as StatusResponse,
+    HandlerResponse as Response,
+    RESPONSE_TYPE
+)
+
+from .utils.helpers import *
+import requests
+import pandas as pd
+from mindsdb_sql import parse_sql, get_lexer_parser
+
+
+class EventStoreDB(DatabaseHandler):
+    """
+    Handler for EventStoreDB
+    The handler uses the Atom Pub Over HTTP of EventStoreDB.
+    This means that you need to enable AtomPuvOverHTTP on your EventStoreDB instance if you are using v20+
+
+    Why not gRPC? At the moment we cannot use https://pypi.org/project/esdbclient/ which uses the gRPC endpoint
+    because mysql-connector-python 8.0.32 requires protobuf<=3.20.3,>=3.11.0, but esdbclient
+    requires protobuf 4.22.1.
+
+    Why not TCP? At the moment https://github.com/epoplavskis/photon-pump only works in insecure mode and
+    configuration is limited.
+
+    Third reason, there is no official Python client at the moment of writing of this handler.
+    But once there is better ESDB Python support for gRPC, we should move this integration from AtomPub to gRPC.
+    """
+
+    name = 'eventstoredb'
+    # defaults to an insecure localhost single node
+    scheme = 'http'
+    host = 'localhost'
+    port = '2113'
+    is_connected = None
+    basic_url = ""
+    read_batch_size = 500  # should be adjusted based on use case

```

Comment:
Do we need this values if we are setting them up in the constructor?

Reply:
It's to support default minimal configuration for example:
CREATE DATABASE insecuredatabase
WITH ENGINE = "eventstoredb",
PARAMETERS = {
    "host": "localhost"
  };

Reply:
For read_batch_size, normally it is of size 20 (because it is normally used by EventStoreDB's Web UI where it lists only 20 events per page) but for MindsDB's case I have found that it much better to set this to 500 for performance reasons. This should only be changed if every event is really big but that is not a recommended practice in EventStoreDB.